pkg/virt-handler/cgroup/cgroup.go

```diff
@@ -0,0 +1,255 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021
+ *
+ */
+
+package cgroup
+
+//go:generate mockgen -source $GOFILE -package=$GOPACKAGE -destination=generated_mock_$GOFILE
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"reflect"
+	"strings"
+
+	"github.com/opencontainers/runc/libcontainer/cgroups"
+	"github.com/opencontainers/runc/libcontainer/cgroups/devices"
+	"github.com/opencontainers/runc/libcontainer/cgroups/ebpf"
+	"github.com/opencontainers/runc/libcontainer/cgroups/ebpf/devicefilter"
+	"github.com/opencontainers/runc/libcontainer/cgroups/fscommon"
+	"github.com/opencontainers/runc/libcontainer/configs"
+
+	"golang.org/x/sys/unix"
+)
+
+var (
+	errNoPreciseMatch = errors.New("resulting devices cgroup doesn't precisely match target")
+	errNoMatch        = errors.New("resulting devices cgroup doesn't match target mode")
+
+	basePath   = "/sys/fs/cgroup"
+	procFormat = "/proc/%d/cgroup"
+
+	isCgroup2UnifiedMode         = cgroups.IsCgroup2UnifiedMode
+	loadAttachCgroupDeviceFilter = ebpf.LoadAttachCgroupDeviceFilter
+)
+
+func ControllerPath(controller string) string {
+	if isCgroup2UnifiedMode() {
+		return basePath
+	}
+	return filepath.Join(basePath, controller)
+}
+
+func CPUSetPath() string {
+	if isCgroup2UnifiedMode() {
```

Comment:
Like in my other comment regarding to `loadAttachCgroupDeviceFilter` it would be great if we can have a clear side-effect free solution for testing.